pkg/gds/models/v1/models.go

```diff
@@ -124,6 +124,56 @@ func AppendAuditLog(vasp *pb.VASP, entry *AuditLogEntry) (err error) {
 	return nil
 }
 
+// GetCertReqIDs returns the list of associated CertificateRequest IDs for the VASP record.
+func GetCertReqIDs(vasp *pb.VASP) (_ []string, err error) {
+	// If the extra data is nil, return nil (no certificate requests).
+	if vasp.Extra == nil {
+		return nil, nil
+	}
+
+	// Unmarshal the extra data field on the VASP.
+	extra := &GDSExtraData{}
+	if err = vasp.Extra.UnmarshalTo(extra); err != nil {
+		return nil, err
+	}
+	return extra.GetCertificateRequests(), nil
+}
+
+// AppendCertReqID adds the certificate request ID to the VASP if its not already added.
+func AppendCertReqID(vasp *pb.VASP, certreqID string) (err error) {
+	// Entry must be non-nil.
+	if certreqID == "" {
+		return errors.New("cannot append empty certificate request ID to extra")
+	}
+
+	// Unmarshal previous extra data.
+	extra := &GDSExtraData{}
+	if vasp.Extra != nil {
+		if err = vasp.Extra.UnmarshalTo(extra); err != nil {
+			return fmt.Errorf("could not deserialize previous extra: %s", err)
+		}
+	} else {
+		extra.CertificateRequests = make([]string, 0, 1)
+	}
+
+	// Do not allow duplicate certificate requests to be appended
+	for _, containsID := range extra.CertificateRequests {
+		if certreqID == containsID {
+			// Do not return an error
```

Comment:
Do we expect this to happen during normal operation or should we be logging a warning?

Reply:
It's mostly just a convenience to make sure we don't do anything wrong in the migration - Append is only being called once in the code right now so I'm not too worried about it. Great question though, thank you!